code/lib/preview-api/src/modules/store/csf/prepareStory.ts

```diff
@@ -149,6 +154,25 @@ export function prepareStory<TRenderer extends Renderer>(
   };
 }
 
+export const getTriangularNumber = (seed: number) => (seed * (seed + 1)) / 2;
+
+export const delayForCanvas = async (canvasElement: HTMLElement) => {
+  let backOff = 0;
+  while (!canvasElement.innerHTML) {
+    backOff += 1;
+    if (backOff > 100) {
+      throw new Error(
```

Comment:
I'm not certain `throw` is the right answer for this code base. If you'd prefer some sort of boolean return and play or not play based on that let me know.